proxies/dotnet/Controllers/User.cs

```diff
@@ -0,0 +1,40 @@
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json.Linq;
+
+// using Newtonsoft.Json;
+using System.Text.Json;
+
+namespace dotnet.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class UserController : ControllerBase
+{
+
+    private readonly ILogger<UserController> _logger;
+    public static Dictionary<string, User> users = new Dictionary<string, User>();
+
+    public UserController(ILogger<UserController> logger)
+    {
+        _logger = logger;
+    }
+
+    [HttpPost]
+    public IActionResult CreateUser(User user)
+    {
+        var serializedUser = JsonSerializer.Deserialize<object>(JsonSerializer.Serialize(user));
+        var result = new {entityType = "user", body = serializedUser};
+
+        var userId = users.Count;
+        users[userId.ToString()] = user;
+
+        HttpContext.Response.Headers.Add("Location", "user/" + userId);
+
+        var res = new ContentResult();
+        res.ContentType = "application/json";
+        res.StatusCode = (int)HttpStatusCode.Created;
+        res.Content = JsonSerializer.Serialize(result);
```

Comment:
you should be able to just return result here rather than these four lines

Reply:
part of the ticket is to have `201` returned since it's creating a new resource

Reply:
does it not do 201 by default on POST?

even if not, you only then need the line that sets the status code. The content type header and JSON body are set automatically when you return a serializable object

Reply:
it does `200` by default, but I changed it to just change the response status code like so: 
```
Response.StatusCode = (int)HttpStatusCode.Created;
```